processing/src/main/java/org/apache/druid/query/expression/LookupExprMacro.java

```diff
@@ -69,7 +70,9 @@ public Expr apply(final List<Expr> args)
         lookupExtractorFactoryContainerProvider,
         lookupName,
         false,
-        null,
+        replaceMissingValueWith != null && replaceMissingValueWith.isLiteral()
+        ? (String) replaceMissingValueWith.getLiteralValue()
```

Comment:
I suppose since other types can also be literals it might be safer to use `Evals.asString` instead of casting

Reply:
fixed